packages/atomic/src/components/common/generated-answer/show-button.tsx

```diff
@@ -0,0 +1,37 @@
+import {FunctionalComponent, h} from '@stencil/core';
+import {i18n} from 'i18next';
+import ArrowBottomIcon from '../../../images/arrow-bottom-rounded.svg';
+import ArrowTopIcon from '../../../images/arrow-top-rounded.svg';
+import {Button} from '../button';
+
+interface ShowButtonProps {
+  i18n: i18n;
+  isCollapsed: boolean;
+  onClick: () => void;
+  isStreaming?: boolean;
+  collapsible?: boolean;
+}
+
+export const ShowButton: FunctionalComponent<ShowButtonProps> = (props) => {
+  const btnClasses = 'hidden items-center hover:bg-transparent';
+
+  return (
+    <Button
+      style="text-primary"
+      part="answer-show-button"
+      class={`${btnClasses} ${props.collapsible ? (props.isStreaming ? 'flex invisible' : 'flex') : 'hidden'}`}
```

Comment:
[clsx](https://www.npmjs.com/package/clsx) weights 8.5 kb.  Might worth a shot here.

```
const btnClasses = 'hidden items-center hover:bg-transparent';
const streamingClasses = props.isStreaming ? 'flex invisible' : 'flex'
[...]
class={clsx([btnClasses, streamingClasses, {hidden: props.collapsible}])}
```
Perfect for soft brain like me.

I get why we wouldn't want to add another lib in a client facing bundle. But I would argue that this is pretty small, and probably already used in most client side app nowaday when dlded 17 millions times a week.

Reply:
I personally disagree; clsx would add 0 value for the end user and would only make our source code slightly more readable. It'd be one more dependency to maintain. I don't think it's worth it.

Reply:
This is exactly the code readability issue that I wanna stress here. I would argue that happy devs that are less inclined to do mistakes because they have to maintain nested ternaries are not 0 value for the end user. But I will not fight over it.

Here I would at least try to get rid of the imbedded ternary in the prop.

Reply:
> Here I would at least try to get rid of the imbedded ternary in the prop.

Yes ! Also, `hidden` is there 2 times, but gets overriden (?) if `collapsible` ? 
This certainly smells fishy and not optimal. Please rework into something more manageable 🙏 .

As for adding a lib, the reasons for adding one in the bundler needs to be "bullet proof" and a "no brainer" in the context of Atomic.

I personally think we can very easily achieve what `clsx` does with just proper string building "discipline". I've looked at the doc of clsx and I must say I would not know what this does exactly: 

```
clsx('foo', [1 && 'bar', { baz:false, bat:null }, ['hello', ['world']]], 'cya');
```

VS just a standard string concatenation (without ternary)

Reply:
:+1: Thank you both :+1: